nemo/collections/speechlm/recipes/pipeline.py

```diff
@@ -0,0 +1,170 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from omegaconf import DictConfig, OmegaConf
+
+from nemo import lightning as nl
+from nemo.collections.common.tokenizers import AutoTokenizer
+from nemo.collections.speechlm.api import _setup, finetune
+from nemo.collections.speechlm.data.audio_to_text_module import AudioToTextDataModule
+from nemo.collections.speechlm.models.speech_to_text_llm_model import SpeechToTextLLM, SpeechToTextLLMConfig
+from nemo.collections.speechlm.modules.asr_module import ASRModuleConfig
+from nemo.collections.speechlm.modules.modality_adapter import ModalityAdapterConfig
+from nemo.collections.speechlm.utils import SpeechToTextLLMPEFT, get_object_list_from_config
+from nemo.core.classes.common import Serialization, typecheck
+from nemo.lightning.pytorch.callbacks import PreemptionCallback
+from nemo.utils import logging
+from nemo.utils.exp_manager import StatelessTimer, TimingCallback
+
+
+def speech_to_text_llm_train(cfg: DictConfig):
```

Comment:
Pretty cool ! If i read this correctly, we can reuse this one function for most SpeechToTextLLM while swapping out the underlying model with hydra target?